src/components/product.js

```diff
@@ -766,4 +773,10 @@ export default class Product extends Component {
     this.selectedVariant = selectedVariant;
     return model;
   }
+
+
+  imageAltText(string) {
```

Comment:
@spencercanner are the `get image()` tests [here](https://github.com/Shopify/buy-button-js/blob/master/test/unit/product/product-component.js#L1145)?

Reply:
We favour mocking out internal function calls where possible, as this allows us to test functions in isolation. Although `Product#imageAltText` is a simple function, we'd rather have it mocked out in `get image()` tests for consistency. Another benefit of doing so is that if the behaviour of `Product#imageAltText` were to change, we would only need to update its tests and not have to potentially update all the image related tests. Here's an [example](https://github.com/Shopify/buy-button-js/blob/master/test/unit/product/product-component.js#L611) of a test block that mocks out an internal function.

Reply:
@spencercanner I'm really sorry but it is a bit unclear to me how I should proceed with regards to stubbing out this private, `imageForSize`, method. In my testing experience which is primarily with Ruby and Rspec, I would stub a `dependency` like an API call and typically let the private methods run on their own accord -- especially if they are just doing some string manipulation. If you want to step into this PR and enforce how you want the code to look like with regards to this stub you have in mind, please be my guest. 

I did, however, make an attempt to stub out `imageForSize` as I think you suggested, but I run into some errors:

Here's how I would stub:
```javascript
 describe('image', () => {
        let imageForSizeStub;
        let imageAltTextStub;

        beforeEach(async () => {
          await product.init(testProductCopy);
          imageForSizeStub = sinon.stub(product.props.client.image.helpers, 'imageForSize').callsFake((image, dimensions) => {
            return dimensions;
          });

          imageAltTextStub = sinon.stub(product, 'imageAltText');
        });

        afterEach(() => {
          imageForSizeStub.restore();
          imageAltTextStub.restore();
        });
```

and here are the errors:
```
1. returns object with id, src, srcLarge, and srcOriginal from selected image if selected image exists
2. returns object with id, src, srcLarge, and srcOriginal from first image in model if selected variant does not have an image
3. returns object with id, src, srcLarge, and srcOriginal from selected variant as default
```
The tests are complaining that the `expectedObject` should contain an `altText` value that matches the key value pair of `product.image`, which of course it does not because I stubbed out the private method `imageAltText`. I think in order to stub successfully I would need to stub out `imageAltText` to return whatever value the test expects for `altText` in each of the above tests.

Having said all that, I have added the tests in a `describe` block for `imageAltText()` so we do have some test coverage on that private method.